lib/global_id/uri/gid.rb

```diff
@@ -154,6 +157,13 @@ def validate_component(component)
           "Expected a URI like gid://app/Person/1234: #{inspect}"
       end
 
+      def validate_model_id(model_id)
+        return model_id unless model_id.blank?
+
+        raise MissingModelId, "Unable to create a global id for #{model_name} " \
```

Comment:
You're in a new method scope, so `model_name` doesn't exist as a variable. It's calling the `model_name` reader method which evaluates to `@model_name`, which is assigned two lines after we validate :wink: